ecommerce-ui/src/NotAirBnB.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import Rentals from './Rentals';
+import Cart from './Cart';
+import airbnbs from './airbnbs.json';
+
+class NotAirBnB extends Component {
+    state = {
+        rentals: airbnbs,
+        selectedRentals: [],
+        cartTotal: 0
+    }
+    
+    addRental = (index) => {
+        const selectedRental = this.state.rentals[index];
+        const currentTotal = this.state.cartTotal;
+        const rentalCost = selectedRental.payment.cost;
+        const newTotal = currentTotal + rentalCost;
+
+        if (this.state.selectedRentals.includes(selectedRental)) {
+            alert('You have already selected this rental.');
+            return;
+        }
+
+        this.setState(prevState => {
+            return {
+              selectedRentals: [...prevState.selectedRentals, selectedRental],
+              cartTotal: newTotal
+            };
+        });
+    }
+
+    
+    deleteRental = (index) => {
+        const selectedRental = this.state.rentals[index];
```

Comment:
Be sure to use this.state.selectedRentals[index]
the index being returned corresponds to that item and not the one in this.state.rentals. 
You can test this out by adding rentals to your cart in a reverse order (right to left) and deleting them. The totals will not subtract correctly because the wrong array is being targeted.